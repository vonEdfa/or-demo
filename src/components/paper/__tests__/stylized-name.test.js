import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import 'jest-styled-components';
import TestRenderer from 'react-test-renderer';

import StylizedName from '../stylized-name';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('StylizedName', () => {
  it('renders with or without props', () => {
    act(() => {
      render(<StylizedName />, container);
    });
    expect(container.textContent).toBe('Name' + 'Surname');

    act(() => {
      render(<StylizedName firstName="Hello" />, container);
    });
    expect(container.textContent).toBe('Hello' + 'Surname');

    act(() => {
      render(<StylizedName surName="World" />, container);
    });
    expect(container.textContent).toBe('Name' + 'World');

    act(() => {
      render(<StylizedName firstName="Hello" surName="World" />, container);
    });
    expect(container.textContent).toBe('Hello' + 'World');
  });

  it('renders correctly', () => {
    act(() => {
      render(<StylizedName />, container);
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();

    act(() => {
      render(<StylizedName firstName="Hello" surName="World" />, container);
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();

    act(() => {
      render(<StylizedName firstName="Hello" surName="world" />, container);
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('uses the correct styles', () => {
    const withCapitalSurname = TestRenderer.create(<StylizedName firstName="Name" surName="Surname" />);
    expect(withCapitalSurname.toJSON).toMatchSnapshot();
    expect(withCapitalSurname.toTree().rendered.props.extraPadding).toBe(true);

    const withCamelCaseSurname = TestRenderer.create(<StylizedName firstName="Name" surName="von Surname" />);
    expect(withCamelCaseSurname.toJSON()).toMatchSnapshot();
    expect(withCamelCaseSurname.toTree().rendered.props.extraPadding).toBe(false);
  });
});
