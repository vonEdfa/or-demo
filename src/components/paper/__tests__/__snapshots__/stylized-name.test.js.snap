// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StylizedName renders correctly 1`] = `
"<div class=\\"sc-bdVaJa czRHzE\\">
  <h1 class=\\"sc-bwzfXH CmELq\\">Name</h1>
  <h1 class=\\"sc-htpNat eGCyKc\\">Surname</h1>
</div>"
`;

exports[`StylizedName renders correctly 2`] = `
"<div class=\\"sc-bdVaJa czRHzE\\">
  <h1 class=\\"sc-bwzfXH CmELq\\">Hello</h1>
  <h1 class=\\"sc-htpNat eGCyKc\\">World</h1>
</div>"
`;

exports[`StylizedName renders correctly 3`] = `
"<div class=\\"sc-bdVaJa bexPbI\\">
  <h1 class=\\"sc-bwzfXH edVCEj\\">Hello</h1>
  <h1 class=\\"sc-htpNat eGCyKc\\">world</h1>
</div>"
`;

exports[`StylizedName uses the correct styles 1`] = `[Function]`;

exports[`StylizedName uses the correct styles 2`] = `
.c0 {
  position: relative;
}

.c0:nth-child(2) {
  margin-top: 2.5em;
}

.c1 {
  padding: 0;
  margin: 0;
  line-height: 1em;
  position: absolute;
  top: -0.2em;
  left: 0;
  text-transform: uppercase;
  font-family: 'Arimo',sans-serif;
  font-weight: normal;
  font-size: 12pt !important;
  color: #a0a5a7;
  z-index: 10;
}

.c2 {
  padding: 0;
  margin: 0;
  font-size: 70pt !important;
  line-height: 0.74em;
  font-family: 'Roboto Slab',serif;
  font-weight: normal;
  font-size: 72pt;
  position: relative;
  left: -0.01em;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Name
  </h1>
  <h1
    className="c2"
  >
    von Surname
  </h1>
</div>
`;
